src/pke/lib/scheme/bgvrns/bgvrns-parametergeneration.cpp

```diff
@@ -194,6 +194,71 @@ std::pair<std::vector<NativeInteger>, uint32_t> ParameterGenerationBGVRNS::compu
 
 }
 
+void ParameterGenerationBGVRNS::InitializeFloddingDgg(
+  std::shared_ptr<CryptoParametersBase<DCRTPoly>> cryptoParams, 
+  usint numPrimes,
+  enum KeySwitchTechnique ksTech,
+  enum ProxyReEncryptionMode PREMode) const {
+  
+  const auto cryptoParamsBGVRNS =
+      std::static_pointer_cast<CryptoParametersBGVRNS>(cryptoParams);
+
+  //compute the flooding distribution parameter based on the security mode for pre
+  double stat_sec=30; //statistical security parameter for noise flooding
+
+  //get the re-encryption level and set the level after re-encryption
+  usint ringDimension = cryptoParamsBGVRNS->GetElementParams()->GetRingDimension();
+  double sigma = cryptoParamsBGVRNS->GetDistributionParameter();
+  double alpha = cryptoParamsBGVRNS->GetAssuranceMeasure();
+  uint32_t r = cryptoParamsBGVRNS->GetRelinWindow();
+  double log2q = log2(cryptoParamsBGVRNS->GetElementParams()->GetModulus().ConvertToDouble());
+
+  double B_e = sqrt(alpha)*sigma;
+  uint32_t auxBits = DCRT_MODULUS::MAX_SIZE;
+  double Bkey = (cryptoParamsBGVRNS->GetMode() == RLWE) ? sigma * sqrt(alpha) : 1;
+
+  if (PREMode == FIXED_NOISE_HRA) {
+    auto &dggflooding = cryptoParamsBGVRNS->GetFloodingDiscreteGaussianGenerator();
+
+    double noise_param = PRE_SD;
+    
+    //set the flooding distribution parameter to the distribution.
+    dggflooding.SetStd(noise_param);
+
+  } else if (PREMode == NOISE_FLOODING_HRA) {
+    //get the flooding discrete gaussian distribution
+    auto &dggflooding = cryptoParamsBGVRNS->GetFloodingDiscreteGaussianGenerator();
+
+    double noise_param=1;
+
+    if (ksTech == BV) {
+      if (r > 0) {
+        noise_param = pow(2,stat_sec) * (1 + 2 * Bkey) * (numPrimes + 1) * (log2q/r+1) * sqrt(ringDimension) * (pow(2,r)-1) * B_e;
+      } else {
+        OPENFHE_THROW(config_error, "Relinwindow value cannot be 0 for BV keyswitching");
+      }
+    } else if (ksTech == HYBRID) {
+      if (r == 0) {
+        double numTowersPerDigit = cryptoParamsBGVRNS->GetNumPerPartQ();
+        int numDigits = cryptoParamsBGVRNS->GetNumPartQ();
+        noise_param = numTowersPerDigit * numDigits * sqrt(ringDimension) * B_e * (1 + 2 * Bkey);
+        noise_param += auxBits * (1 + sqrt(ringDimension));
+        noise_param = pow(2,stat_sec) * noise_param;
+      } else {
+        OPENFHE_THROW(config_error, "Relinwindow value can be non-zero only for BV keyswitching");
+      }
+    }
```

Comment:
this "if" statement missing "else" or a comment instead

Reply:
fixed